File: java/src/main/java/mx/iteso/stores/WaterIceCream.java
package mx.iteso.stores;

import mx.iteso.IceCreamStore;
import mx.iteso.icecreams.IceCream;
import mx.iteso.icecreams.water.GuavaWaterIceCream;
import mx.iteso.icecreams.water.MangoWaterIceCream;
import mx.iteso.icecreams.water.PineappleWaterIceCream;
import mx.iteso.icecreams.water.StrawberryWaterIceCream;
import mx.iteso.icecreams.water.NoFlavorWaterIceCream;

/**
 * Para hacer llamada del helado de agua.
 */
public class WaterIceCream extends IceCreamStore {
    /**
     * Create ice cream.
     * @param selectedFlavor the desired flavor.
     * @return IceCream.
     */
    @Override
    protected IceCream iceCreamFlavor(final String selectedFlavor) {
        if (selectedFlavor.equals("Guava")) {
            return new GuavaWaterIceCream();
        } else if (selectedFlavor.equals("Mango")) {
            return new MangoWaterIceCream();
        } else if (selectedFlavor.equals("Pineapple")) {
            return new PineappleWaterIceCream();
        } else if (selectedFlavor.equals("Strawberry")) {
            return new StrawberryWaterIceCream();
        }
        return new NoFlavorWaterIceCream();
    }
}

File: csharp/factory-method/WaterIceCream.cs
using factorymethod.main.icecreams;
using factorymethod.main.icecreams.water;

namespace factorymethod.main.stores
{
    internal class WaterceCreamStore : IceCreamStore
    {
        override
            protected IceCream createIceCream(string flavor)
        {
            if (flavor.Equals("Strawberry"))
            {
                return new StrawberryWaterIceCream();
            }

            if (flavor.Equals("Pineapple"))
            {
                return new PineappleWaterIceCream();
            }
            if (flavor.Equals("Mango"))
            {
                return new MangoWaterIceCream();
            }
            return flavor.Equals("Guava") ? new GuavaWaterIceCream() : null;
        }

    }
}
}